src/logging.c

```diff
@@ -466,12 +477,12 @@ static void create_format_string(
         // copy the decoded character
         g_rcutils_logging_output_format_string[dest_buffer_index] = expected_char[0];
```

Comment:
if expected_char is `\x1b`, do we need to copy the index 1 and 2 here?

Reply:
No we don't. The `\x1b` is only 4 chars long in its written literal (input) representation. When parsed as a escape sequence it is only 2 chars long. Similar to `\n` which is 2 chars long in the written representation but only 1 char long in its representation as a newline command. I added a comment to make this more clear.